handlers.go

```diff
@@ -494,13 +503,15 @@ func (b *BouncerHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	// HACKS
-	// If the user is coming from windows xp or vista, send a sha1
-	// signed product
-	// HACKS
+	// If the user is coming from windows xp or vista, send a sha1 signed product.
 	if reqParams.OS == "win" && isWinXpClient {
 		reqParams.Product = sha1Product(reqParams.Product)
 	}
-	// If the user is an "old" stub installer, send a pre-2024-cert-rotation product
+	// If the user is coming from windows 7/8/8.1, send the latest compatible ESR product.
+	if (reqParams.OS == "win" || reqParams.OS == "win64") && shouldReturnESR115 {
```

Comment:
We should probably have a check on the incoming reqParams.Product, since I guess we don't want to send the exe installer if the request was for msi, etc.

Reply:
mmm, maybe? But what kind of check? Would our msi work on those OSes?

Reply:
It probably would work, but I suggest we just leave reqParams.Product alone if it's not the full installer (or the stub, maybe).

Reply:
I filtered out `-msi` products instead.